Adapt areglogger to integrate in Qt

**Description:**

The `areglogger` project is integrated with Qt based application (`areg-sdk-tools` project), but it is not linked if the class triggers signal that contains `TEArrayList` class or other container based on STL.
These are error messages:

```text
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\xutility:5540: error: C3889: call to object of class type 'std::equal_to<void>': no matching call operator found
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\xutility(5540): error C3889: call to object of class type 'std::equal_to<void>': no matching call operator found
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\xutility(726): note: could be 'unknown-type std::equal_to<void>::operator ()(_Ty1 &&,_Ty2 &&) noexcept(<expr>) const'
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\xutility(5540): note: Failed to specialize function template 'unknown-type std::equal_to<void>::operator ()(_Ty1 &&,_Ty2 &&) noexcept(<expr>) const'
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\xutility(5540): note: With the following template arguments:
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\xutility(5540): note: '_Ty1=const _Ty &'
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\xutility(5540): note: '_Ty2=const _Ty &'
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\xutility(728): note: binary '==': 'const _Ty' does not define this operator or a conversion to a type acceptable to the predefined operator
-----
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\utility:503: could be 'bool std::operator ==(const std::pair<_Ty1,_Ty2> &,const std::pair<_Uty1,_Uty2> &)'
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\utility(503): note: could be 'bool std::operator ==(const std::pair<_Ty1,_Ty2> &,const std::pair<_Uty1,_Uty2> &)'
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\xutility(5540): note: the template instantiation context (the oldest one first) is
C:\projects\areg-sdk-tools\build\Desktop_Qt_6_8_0_MSVC2022_64bit-Debug\lusan_autogen\CMN6AJKCLM/moc_LogObserverClient.cpp(128): note: see reference to variable template 'const QtPrivate::QMetaTypeInterface *const qt_incomplete_metaTypeArray<`anonymous namespace'::qt_meta_stringdata_CLASSLogObserverClientENDCLASS_t,QtPrivate::TypeAndForceComplete<LogObserverClient,std::integral_constant<bool,1> >,QtPrivate::TypeAndForceComplete<void,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<bool,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<String const &,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<unsigned short,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<void,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<bool,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<String const &,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<String const &,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<String const &,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<void,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<bool,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<String const &,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<unsigned short,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<void,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<bool,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<void,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<String const &,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<void,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<void,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<TEArrayList<NEService::sServiceConnectedInstance> const &,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<void,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<TEArrayList<NEService::sServiceConnectedInstance> const &,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<void,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<TEMap<unsigned __int64,NEService::sServiceConnectedInstance> const &,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<void,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<unsigned __int64,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<sLogScope const *,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<int,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<void,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<unsigned __int64,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<sLogScope const *,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<int,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<void,std::integral_constant<bool,0> >,QtPrivate::TypeAndForceComplete<sLogMessage const *,std::integral_constant<bool,0> > >[35]' being compiled
C:\DevTools\Qt\6.8.0\msvc2022_64\include\QtCore/qmetatype.h(2689): note: see reference to function template instantiation 'const QtPrivate::QMetaTypeInterface *QtPrivate::qTryMetaTypeInterfaceForType<`anonymous-namespace'::qt_meta_stringdata_CLASSLogObserverClientENDCLASS_t,QtPrivate::TypeAndForceComplete<const TEArrayList<NEService::sServiceConnectedInstance> &,std::false_type>>(void)' being compiled
C:\DevTools\Qt\6.8.0\msvc2022_64\include\QtCore/qmetatype.h(2633): note: see reference to class template instantiation 'QtPrivate::is_complete<QtPrivate::qTryMetaTypeInterfaceForType::Tz,`anonymous-namespace'::qt_meta_stringdata_CLASSLogObserverClientENDCLASS_t>' being compiled
C:\DevTools\Qt\6.8.0\msvc2022_64\include\QtCore/qmetatype.h(890): note: see reference to class template instantiation 'QtPrivate::detail::is_complete_helper<TEArrayList<NEService::sServiceConnectedInstance>,ODR_VIOLATION_PREVENTER>' being compiled 


----

C:\DevTools\Qt\6.8.0\msvc2022_64\include\QtCore\qmetatype.h:885: while compiling class template member function 'std::integral_constant<bool,sizeof(U)!=0> QtPrivate::detail::is_complete_helper<TEArrayList<NEService::sServiceConnectedInstance>,ODR_VIOLATION_PREVENTER>::check(U *)'
C:\DevTools\Qt\6.8.0\msvc2022_64\include\QtCore\qmetatype.h:883: see reference to class template instantiation 'TEArrayList<NEService::sServiceConnectedInstance>' being compiled
C:\DevTools\Qt\6.8.0\msvc2022_64\include\QtCore/qmetatype.h(883): note: see reference to class template instantiation 'TEArrayList<NEService::sServiceConnectedInstance>' being compiled
C:\projects\areg-sdk-tools\build\Desktop_Qt_6_8_0_MSVC2022_64bit-Debug\packages\areg-sdk-src\framework\areg/base/TEArrayList.hpp(605): note: while compiling class template member function 'bool TEArrayList<NEService::sServiceConnectedInstance>::operator ==(const TEArrayList<NEService::sServiceConnectedInstance> &) const'
C:\DevTools\Qt\6.8.0\msvc2022_64\include\QtCore/qmetatype.h(2360): note: see the first reference to 'TEArrayList<NEService::sServiceConnectedInstance>::operator ==' in 'QtPrivate::QEqualityOperatorForType<T,true>::equals'
C:\DevTools\Qt\6.8.0\msvc2022_64\include\QtCore\qmetatype.h:2522: see the first reference to 'QtPrivate::QEqualityOperatorForType<T,true>::equals'
C:\projects\areg-sdk-tools\build\Desktop_Qt_6_8_0_MSVC2022_64bit-Debug\packages\areg-sdk-src\framework\areg\base\TEArrayList.hpp:606: see reference to function template instantiation 'bool std::operator ==<VALUE,std::allocator<VALUE>>(const std::vector<VALUE,std::allocator<VALUE>> &,const std::vector<VALUE,std::allocator<VALUE>> &)' being compiled
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\vector:2260: see reference to function template instantiation 'bool std::equal<const _Ty*,const _Ty*>(const _InIt1,const _InIt1,const _InIt2)' being compiled
C:\Program Files\Microsoft Visual Studio\2022\Community\VC\Tools\MSVC\14.43.34808\include\xutility:5557: see reference to function template instantiation 'bool std::equal<_InIt1,_InIt2,std::equal_to<void>>(const _InIt1,const _InIt1,const _InIt2,_Pr)' being compiled
```

this should be fixed, so that Qt based project can be build using `areg` and/or `areglogger` libraries.

--- a/framework/areg/base/IEIOStream.hpp
+++ b/framework/areg/base/IEIOStream.hpp
@@ -527,16 +527,16 @@ inline IEOutStream & operator << (IEOutStream & stream, const wchar_t * output)
 template<typename CharType>
 inline IEOutStream& operator << (IEOutStream& stream, const std::basic_string<CharType>& output)
 {
-    constexpr unsigned int single = sizeof(CharType);
-    stream.write(reinterpret_cast<const unsigned char*>(output.c_str()), (output.length() + 1) * single);
+    constexpr unsigned int single = static_cast<unsigned int>(sizeof(CharType));
+    stream.write(reinterpret_cast<const unsigned char*>(output.c_str()), static_cast<unsigned int>(output.length() + 1) * single);
     return stream;
 }
 
 template<typename CharType>
 inline IEOutStream& operator << (IEOutStream& stream, const std::basic_string_view<CharType>& output)
 {
-    constexpr unsigned int single = sizeof(CharType);
-    stream.write(reinterpret_cast<const unsigned char*>(output.data()), (output.length() + 1) * single);
+    constexpr unsigned int single = static_cast<unsigned int>(sizeof(CharType));
+    stream.write(reinterpret_cast<const unsigned char*>(output.data()), static_cast<unsigned int>(output.length() + 1) * single);
     return stream;
 }
 

--- a/framework/areg/component/NEService.hpp
+++ b/framework/areg/component/NEService.hpp
@@ -1070,9 +1070,9 @@ namespace NEService
         //!< The connection timestamp
         TIME64                      ciTimestamp { 0 };
         //!< The name of the application
-        String                      ciInstance  { "" };
+        std::string                 ciInstance  { "" };
         //!< The optional file location
-        String                      ciLocation  { "" };
+        std::string                 ciLocation  { "" };
     };
 
     /**

--- a/framework/aregextend/db/private/LogSqliteDatabase.cpp
+++ b/framework/aregextend/db/private/LogSqliteDatabase.cpp
@@ -452,7 +452,7 @@ bool LogSqliteDatabase::logInstanceConnected(const NEService::sServiceConnectedI
     char msg[MSG_LEN];
     String::formatString( msg, MSG_LEN, "The %u-bit instance [ %s ] with cookie [ %llu ] is connected at time [ %s ]"
                         , static_cast<uint32_t>(instance.ciBitness)
-                        , instance.ciInstance.getString()
+                        , instance.ciInstance.c_str()
                         , static_cast<uint64_t>(instance.ciCookie)
                         , timestamp.formatTime().getString());
 
@@ -476,8 +476,8 @@ bool LogSqliteDatabase::logInstanceConnected(const NEService::sServiceConnectedI
                         , static_cast<uint64_t>(instance.ciCookie)
                         , static_cast<uint32_t>(instance.ciSource)
                         , static_cast<uint32_t>(instance.ciBitness)
-                        , instance.ciInstance.getString()
-                        , instance.ciLocation.getString()
+                        , instance.ciInstance.c_str()
+                        , instance.ciLocation.c_str()
                         , static_cast<uint64_t>(instance.ciTimestamp)
                         , static_cast<uint64_t>(timestamp.getTime())
                         );

--- a/framework/areglogger/client/LogObserverBase.hpp
+++ b/framework/areglogger/client/LogObserverBase.hpp
@@ -22,14 +22,15 @@
 #include "areglogger/client/LogObserverSwitches.h"
 #include "areglogger/client/LogObserverApi.h"
 
-#include "areg/base/String.hpp"
-#include "areg/base/TEArrayList.hpp"
 #include "areg/component/NEService.hpp"
+
+#include <map>
+#include <string>
+#include <vector>
 
 /************************************************************************
  * Dependencies.
  ************************************************************************/
-class String;
 class SharedBuffer;
 
 /**
@@ -71,7 +72,7 @@ public:
      *                      If string is empty, uses `./config/areg.init` relative path.
      * \return  Returns true if successfully initialized.
      **/
-    bool initialize(const String& configFile = String::EmptyString);
+    bool initialize(const std::string& configFile = String::EmptyString);
 
     /**
      * \brief   Releases the log observer and disconnects from log collector.
@@ -91,7 +92,7 @@ public:
      * \return  Returns true if succeeded to trigger connection. The physical connection might not be established yet.
      *          The physical connection is established when triggers callback of type FuncServiceConnected.
      **/
-    bool connect(const String& address, uint16_t portNr, const String& dbLocation);
+    bool connect(const std::string& address, uint16_t portNr, const std::string& dbLocation);
 
     /**
      * \brief   Disconnects from the log collector service.
@@ -127,7 +128,7 @@ public:
      *                      The path with mask like `log_%time%.sqlog` are allowed.
      * \return  Returns true if processed with success. Otherwise, returns false.
      **/
-    bool restart(const String& dbLocation = String::EmptyString);
+    bool restart(const std::string& dbLocation = String::EmptyString);
 
     /**
      * \brief   Returns true if the log observer is initialized.
@@ -147,7 +148,7 @@ public:
     /**
      * \brief   Returns the IP address to connect to the log collector service.
      **/
-    const String& getLoggerAddress(void) const;
+    const std::string& getLoggerAddress(void) const;
 
     /**
      * \brief   Returns the TCP port number to connect to the log collector service.
@@ -162,13 +163,13 @@ public:
     /**
      * \brief   Return the IP address of log collector service set in the configuration.
      **/
-    String getConfigLoggerAddress(void) const;
+    std::string getConfigLoggerAddress(void) const;
 
     /**
      * \brief   Sets the IP address of the log collector service in the configuration.
      * \param   address     The IP address of the log collector service to set.
      **/
-    void setConfigLoggerAddress(const String& address);
+    void setConfigLoggerAddress(const std::string& address);
 
     /**
      * \brief   Return the TCP port number of log collector service set in the configuration.
@@ -186,31 +187,31 @@ public:
      * \param   address     The IP address of the log collector service to set.
      * \param   portNr      The TCP port number of the log collector service to set.
      **/
-    void setConfigLoggerConnection(const String& address, uint16_t portNr);
+    void setConfigLoggerConnection(const std::string& address, uint16_t portNr);
 
     /**
      * \brief   Returns the database path name set in the configuration.
      **/
-    String getConfigLoggerDatabase(void) const;
+    std::string getConfigLoggerDatabase(void) const;
 
     /**
      * \brief   Sets the database path name in the configuration.
      *          The path may contain mask like `log_%time%.sqlog`.
      * \param   dbLocation  The database path to set.
      **/
-    void setConfigLoggerDatabase(const String& dbLocation);
+    void setConfigLoggerDatabase(const std::string& dbLocation);
 
     /**
      * \brief   Returns the path of the currently active logging database. The returned path cannot contain mask.
      **/
-    String getActiveDatabasePath(void) const;
+    std::string getActiveDatabasePath(void) const;
 
     /**
      * \brief   Returns the path of the database set during initialization.
      *          The path may contain mask like `log_%time%.sqlog`.
      * \return  Returns the path of the database set during initialization.
      **/
-    String getInitDatabasePath(void) const;
+    std::string getInitDatabasePath(void) const;
 
 //////////////////////////////////////////////////////////////////////////
 // Actions
@@ -276,7 +277,7 @@ protected:
      * \param   address         The null-terminated string of the IP address of the log collector service set in the configuration file.
      * \param   port            The IP port number of the log collector service set in the configuration file.
      **/
-    virtual void onLogObserverConfigured(bool isEnabled, const String & address, uint16_t port) = 0;
+    virtual void onLogObserverConfigured(bool isEnabled, const std::string & address, uint16_t port) = 0;
 
     /**
      * \brief   The callback of the event triggered when initializing and configuring the observer.
@@ -287,7 +288,7 @@ protected:
      * \param   dbLocation      The relative or absolute path the database. The path may contain a mask.
      * \param   dbUser          The database user to use when log in. If null or empty, the database may not require the user name.
      **/
-    virtual void onLogDbConfigured(bool isEnabled, const String & dbName, const String & dbLocation, const String & dbUser) = 0;
+    virtual void onLogDbConfigured(bool isEnabled, const std::string & dbName, const std::string & dbLocation, const std::string & dbUser) = 0;
 
     /**
      * \brief   The callback of the event triggered when the observer connects or disconnects from the log collector service.
@@ -295,7 +296,7 @@ protected:
      * \param   address         The IP address of the log collector service to connect or disconnect.
      * \param   port            The IP port number of the log collector service to connect or disconnect.
      **/
-    virtual void onLogServiceConnected(bool isConnected, const String & address, uint16_t port) = 0;
+    virtual void onLogServiceConnected(bool isConnected, const std::string & address, uint16_t port) = 0;
 
     /**
      * \brief   The callback of the event trigger when starting or pausing the log observer.
@@ -309,7 +310,7 @@ protected:
      * \brief   The callback of the event triggered when the logging database is created.
      * \param   dbLocation      The relative or absolute path to the logging database.
      **/
-    virtual void onLogDbCreated(const String & dbLocation) = 0;
+    virtual void onLogDbCreated(const std::string & dbLocation) = 0;
 
     /**
      * \brief   The callback of the event triggered when fails to send or receive message.
@@ -320,20 +321,20 @@ protected:
      * \brief   The callback of the event triggered when receive the list of connected instances that make logs.
      * \param   instances   The list of the connected instances.
      **/
-    virtual void onLogInstancesConnect(const TEArrayList< NEService::sServiceConnectedInstance > & instances) = 0;
+    virtual void onLogInstancesConnect(const std::vector< NEService::sServiceConnectedInstance > & instances) = 0;
 
     /**
      * \brief   The callback of the event triggered when receive the list of disconnected instances that make logs.
      * \param   instances   The list of IDs of the disconnected instances.
      * \param   count       The number of entries in the list.
      **/
-    virtual void onLogInstancesDisconnect(const TEArrayList< NEService::sServiceConnectedInstance > & instances) = 0;
+    virtual void onLogInstancesDisconnect(const std::vector< NEService::sServiceConnectedInstance > & instances) = 0;
 
     /**
      * \brief   The callback of the event triggered when connection with the log collector service is lost.
      * \param   instances   The list of disconnected instances.
      **/
-    virtual void onLogServiceDisconnected(const NEService::MapInstances& instances) = 0;
+    virtual void onLogServiceDisconnected(const std::map<ITEM_ID, NEService::sServiceConnectedInstance>& instances) = 0;
 
     /**
      * \brief   The callback of the event triggered when receive the list of the scopes registered in an application.

--- a/framework/areglogger/client/private/LogObserverBase.cpp
+++ b/framework/areglogger/client/private/LogObserverBase.cpp
@@ -15,13 +15,12 @@
  ************************************************************************/
 
 #include "areglogger/client/LogObserverBase.hpp"
-#include "areg/base/String.hpp"
-#include "areg/base/SharedBuffer.hpp"
-
 #include "areglogger/client/private/LoggerClient.hpp"
 #include "areglogger/client/LogObserverApi.h"
 
 #include "areg/base/String.hpp"
+#include "areg/base/SharedBuffer.hpp"
+#include "areg/base/TEArrayList.hpp"
 #include "areg/base/NESocket.hpp"
 
 LogObserverBase* LogObserverBase::_theLogObserver{ nullptr };
@@ -38,9 +37,9 @@ LogObserverBase::~LogObserverBase(void)
     LogObserverBase::_theLogObserver = nullptr;
 }
 
-bool LogObserverBase::initialize(const String& configFile)
+bool LogObserverBase::initialize(const std::string& configFile)
 {
-    return logObserverInitialize(nullptr, configFile.isEmpty() ? nullptr : configFile.getString());
+    return logObserverInitialize(nullptr, configFile.empty() ? nullptr : configFile.c_str());
 }
 
 void LogObserverBase::release(void)
@@ -48,10 +47,10 @@ void LogObserverBase::release(void)
     return logObserverRelease();
 }
 
-bool LogObserverBase::connect(const String& address, uint16_t portNr, const String& dbLocation)
+bool LogObserverBase::connect(const std::string& address, uint16_t portNr, const std::string& dbLocation)
 {
-    return logObserverConnectLogger(  dbLocation.isEmpty() ? nullptr : dbLocation.getString()
-                                    , address.isEmpty() ? nullptr : address.getString()
+    return logObserverConnectLogger(  dbLocation.empty() ? nullptr : dbLocation.c_str()
+                                    , address.empty() ? nullptr : address.c_str()
                                     , portNr);
 }
 
@@ -75,9 +74,9 @@ bool LogObserverBase::stop(void)
     return logObserverStopLogging(true, nullptr);
 }
 
-bool LogObserverBase::restart(const String & dbLocation /*= String::EmptyString*/)
+bool LogObserverBase::restart(const std::string & dbLocation /*= String::EmptyString*/)
 {
-    return logObserverStopLogging(false, dbLocation.isEmpty() ? nullptr : dbLocation.getString());
+    return logObserverStopLogging(false, dbLocation.empty() ? nullptr : dbLocation.c_str());
 }
 
 bool LogObserverBase::isInitialized(void) const
@@ -95,9 +94,9 @@ bool LogObserverBase::isStated(void) const
     return logObserverIsStarted();
 }
 
-const String& LogObserverBase::getLoggerAddress(void) const
+const std::string& LogObserverBase::getLoggerAddress(void) const
 {
-    return LoggerClient::getInstance().getAddress().getHostAddress();
+    return LoggerClient::getInstance().getAddress().getHostAddress().getData();
 }
 
 uint16_t LogObserverBase::getLoggerPort(void) const
@@ -110,12 +109,12 @@ bool LogObserverBase::getConfigLoggerEnabled(void) const
     return logObserverConfigLoggerEnabled();
 }
 
-String LogObserverBase::getConfigLoggerAddress(void) const
+std::string LogObserverBase::getConfigLoggerAddress(void) const
 {
-    return LoggerClient::getInstance().getConfigLoggerAddress();
+    return LoggerClient::getInstance().getConfigLoggerAddress().getData();
 }
 
-void LogObserverBase::setConfigLoggerAddress(const String& address)
+void LogObserverBase::setConfigLoggerAddress(const std::string& address)
 {
     LoggerClient::getInstance().setConfigLoggerConnection(address, NESocket::InvalidPort);
 }
@@ -130,29 +129,29 @@ void LogObserverBase::setConfigLoggerPort(uint16_t portNr)
     LoggerClient::getInstance().setConfigLoggerConnection(String::EmptyString, portNr);
 }
 
-void LogObserverBase::setConfigLoggerConnection(const String& address, uint16_t portNr)
+void LogObserverBase::setConfigLoggerConnection(const std::string& address, uint16_t portNr)
 {
     LoggerClient::getInstance().setConfigLoggerConnection(address, portNr);
 }
 
-String LogObserverBase::getConfigLoggerDatabase(void) const
+std::string LogObserverBase::getConfigLoggerDatabase(void) const
 {
-    return LoggerClient::getInstance().getConfigDatabasePath();
+    return LoggerClient::getInstance().getConfigDatabasePath().getData();
 }
 
-void LogObserverBase::setConfigLoggerDatabase(const String& dbLocation)
+void LogObserverBase::setConfigLoggerDatabase(const std::string& dbLocation)
 {
     LoggerClient::getInstance().setConfigDatabasePath(dbLocation, true);
 }
 
-String LogObserverBase::getActiveDatabasePath(void) const
+std::string LogObserverBase::getActiveDatabasePath(void) const
 {
-    return LoggerClient::getInstance().getActiveDatabasePath();
+    return LoggerClient::getInstance().getActiveDatabasePath().getData();
 }
 
-String LogObserverBase::getInitDatabasePath(void) const
+std::string LogObserverBase::getInitDatabasePath(void) const
 {
-    return LoggerClient::getInstance().getInitialDatabasePath();
+    return LoggerClient::getInstance().getInitialDatabasePath().getData();
 }
 
 bool LogObserverBase::requestInstances(void)
@@ -179,4 +178,3 @@ void LogObserverBase::saveLoggerConfig(void)
 {
     LoggerClient::getInstance().saveConfiguration();
 }
-

--- a/framework/areglogger/client/private/LoggerClient.cpp
+++ b/framework/areglogger/client/private/LoggerClient.cpp
@@ -238,7 +238,7 @@ bool LoggerClient::openLoggingDatabase(const char* dbPath /*= nullptr*/)
     FuncLogDbCreated callback{ mLogDatabase.isOperable() && (mCallbacks != nullptr) ? mCallbacks->evtLogDbCreated  : nullptr};
     if (LogObserverBase::_theLogObserver != nullptr)
     {
-        LogObserverBase::_theLogObserver->onLogDbCreated(mLogDatabase.getDatabasePath());
+        LogObserverBase::_theLogObserver->onLogDbCreated(mLogDatabase.getDatabasePath().getData());
     }
     else if (callback != nullptr)
     {
@@ -341,8 +341,8 @@ void LoggerClient::postReadConfiguration(ConfigManager& config)
 
     if (LogObserverBase::_theLogObserver != nullptr)
     {
-        LogObserverBase::_theLogObserver->onLogObserverConfigured(true, address, port);
-        LogObserverBase::_theLogObserver->onLogDbConfigured(config.getLogEnabled(NELogging::eLogingTypes::LogTypeDatabase), dbName, dbLocation, dbUser);
+        LogObserverBase::_theLogObserver->onLogObserverConfigured(true, address.getData(), port);
+        LogObserverBase::_theLogObserver->onLogDbConfigured(config.getLogEnabled(NELogging::eLogingTypes::LogTypeDatabase), dbName.getData(), dbLocation.getData(), dbUser.getData());
     }
     else
     {
@@ -409,7 +409,7 @@ void LoggerClient::disconnectServiceHost(void)
         FuncInstancesDisconnect callback{ mCallbacks != nullptr ? mCallbacks->evtInstDisconnected : nullptr };
         if (LogObserverBase::_theLogObserver != nullptr)
         {
-            LogObserverBase::_theLogObserver->onLogServiceDisconnected(mInstances);
+            LogObserverBase::_theLogObserver->onLogServiceDisconnected(mInstances.getData());
         }
         else if (callback != nullptr)
         {
@@ -461,7 +461,7 @@ void LoggerClient::connectedRemoteServiceChannel(const Channel& channel)
 
     if (LogObserverBase::_theLogObserver != nullptr)
     {
-        LogObserverBase::_theLogObserver->onLogServiceConnected(true, address, port);
+        LogObserverBase::_theLogObserver->onLogServiceConnected(true, address.getData(), port);
         LogObserverBase::_theLogObserver->onLogObserverStarted(isStarted);
     }
     else
@@ -497,7 +497,7 @@ void LoggerClient::disconnectedRemoteServiceChannel(const Channel& /* channel */
     if (LogObserverBase::_theLogObserver != nullptr)
     {
         LogObserverBase::_theLogObserver->onLogObserverStarted(false);
-        LogObserverBase::_theLogObserver->onLogServiceConnected(false, address, port);
+        LogObserverBase::_theLogObserver->onLogServiceConnected(false, address.getData(), port);
     }
     else
     {

--- a/framework/areglogger/client/private/ObserverMessageProcessor.cpp
+++ b/framework/areglogger/client/private/ObserverMessageProcessor.cpp
@@ -251,12 +251,12 @@ void ObserverMessageProcessor::_clientsConnected(const RemoteMessage& msgReceive
                     inst.liTimestamp = client.ciTimestamp;
                     NEString::copyString( inst.liName
                                         , static_cast<NEString::CharCount>(LENGTH_NAME)
-                                        , client.ciInstance.getString()
-                                        , static_cast<NEString::CharCount>(client.ciInstance.getLength()));
+                                        , client.ciInstance.c_str()
+                                        , static_cast<NEString::CharCount>(client.ciInstance.length()));
                     NEString::copyString( inst.liLocation
                                         , static_cast<NEString::CharCount>(LENGTH_LOCATION)
-                                        , client.ciLocation.getString()
-                                        , static_cast<NEString::CharCount>(client.ciLocation.getLength()));
+                                        , client.ciLocation.c_str()
+                                        , static_cast<NEString::CharCount>(client.ciLocation.length()));
             }
         }
 
@@ -264,7 +264,7 @@ void ObserverMessageProcessor::_clientsConnected(const RemoteMessage& msgReceive
         mLoggerClient.mLock.unlock();
         if (LogObserverBase::_theLogObserver != nullptr)
         {
-            LogObserverBase::_theLogObserver->onLogInstancesConnect(listConnected);
+            LogObserverBase::_theLogObserver->onLogInstancesConnect(listConnected.getData());
         }
         else if (callback != nullptr)
         {
@@ -320,7 +320,7 @@ void ObserverMessageProcessor::_clientsDisconnected(const RemoteMessage& msgRece
         mLoggerClient.mLock.unlock();
         if (LogObserverBase::_theLogObserver != nullptr)
         {
-            LogObserverBase::_theLogObserver->onLogInstancesDisconnect(listDisconnected);
+            LogObserverBase::_theLogObserver->onLogInstancesDisconnect(listDisconnected.getData());
         }
         else if (callback != nullptr)
         {

--- a/framework/logcollector/app/private/LogCollector.cpp
+++ b/framework/logcollector/app/private/LogCollector.cpp
@@ -504,7 +504,7 @@ void LogCollector::_outputInstances( const NEService::MapInstances & instances )
             instances.getAtPosition( pos, cookie, instance);
             unsigned int id{ static_cast<unsigned int>(cookie) };
 
-            console.outputMsg(coord, " %4d. |  %11u  |    %u     |  %s ", i++, id, static_cast<uint32_t>(instance.ciBitness), instance.ciInstance.getString());
+            console.outputMsg(coord, " %4d. |  %11u  |    %u     |  %s ", i++, id, static_cast<uint32_t>(instance.ciBitness), instance.ciInstance.c_str());
             ++ coord.posY;
         }
     }

--- a/framework/logcollector/app/private/LogCollector.cpp
+++ b/framework/logcollector/app/private/LogCollector.cpp
@@ -532,7 +532,7 @@ void LogCollector::_outputInstances( const NEService::MapInstances & instances )
             instances.getAtPosition( pos, cookie, instance);
             unsigned int id{ static_cast<unsigned int>(cookie) };
 
-            printf(" %4d. |  %11u  |    %u     |  %s \n", i++, id, static_cast<uint32_t>(instance.ciBitness), instance.ciInstance.getString());
+            printf(" %4d. |  %11u  |    %u     |  %s \n", i++, id, static_cast<uint32_t>(instance.ciBitness), instance.ciInstance.c_str());
         }
     }
 

--- a/framework/logcollector/service/private/LogCollectorServerService.cpp
+++ b/framework/logcollector/service/private/LogCollectorServerService.cpp
@@ -50,9 +50,9 @@ void LogCollectorServerService::addInstance(const ITEM_ID& cookie, const NEServi
                             , NELogging::LOG_MESSAGE_IZE
                             , "CONNECTED the %u-bit instance [ %s ] with cookie [ %llu ] and location [ %s ]"
                             , static_cast<uint32_t>(instance.ciBitness)
-                            , instance.ciInstance.getString()
+                            , instance.ciInstance.c_str()
                             , instance.ciCookie
-                            , instance.ciLocation.getString());
+                            , instance.ciLocation.c_str());
         NELogging::logAnyMessageLocal(logMsgHello);
 
         mLoggerProcessor.notifyConnectedInstances(getInstances(), NEService::COOKIE_ANY);
@@ -80,9 +80,9 @@ void LogCollectorServerService::removeInstance(const ITEM_ID & cookie)
                             , NELogging::LOG_MESSAGE_IZE
                             , "DISCONNECTED the %u-bit instance [ %s ] with cookie [ %llu ] and location [ %s ]"
                             , static_cast<uint32_t>(instance.ciBitness)
-                            , instance.ciInstance.getString()
+                            , instance.ciInstance.c_str()
                             , instance.ciCookie
-                            , instance.ciLocation.getString());
+                            , instance.ciLocation.c_str());
         NELogging::logAnyMessageLocal(logMsgBye);
 
         listIds.add(instance.ciCookie);

--- a/framework/mcrouter/app/private/MulticastRouter.cpp
+++ b/framework/mcrouter/app/private/MulticastRouter.cpp
@@ -490,7 +490,7 @@ void MulticastRouter::_outputInstances( const NEService::MapInstances & instance
             instances.getAtPosition( pos, cookie, instance);
             unsigned int id{ static_cast<unsigned int>(cookie) };
 
-            printf(" %4d. |  %11u  |    %u     |  %s \n", i++, id, static_cast<uint32_t>(instance.ciBitness), instance.ciInstance.getString());
+            printf(" %4d. |  %11u  |    %u     |  %s \n", i++, id, static_cast<uint32_t>(instance.ciBitness), instance.ciInstance.c_str());
         }
     }
 

--- a/framework/mcrouter/app/private/MulticastRouter.cpp
+++ b/framework/mcrouter/app/private/MulticastRouter.cpp
@@ -462,7 +462,7 @@ void MulticastRouter::_outputInstances( const NEService::MapInstances & instance
             instances.getAtPosition( pos, cookie, instance);
             unsigned int id{ static_cast<unsigned int>(cookie) };
 
-            console.outputMsg(coord, " %4d. |  %11u  |    %u     |  %s ", i++, id, static_cast<uint32_t>(instance.ciBitness), instance.ciInstance.getString());
+            console.outputMsg(coord, " %4d. |  %11u  |    %u     |  %s ", i++, id, static_cast<uint32_t>(instance.ciBitness), instance.ciInstance.c_str());
             ++ coord.posY;
         }
     }

